src/ProductConstructionService/ProductConstructionService.WorkItems/ReplicaWorkItemProcessorStateFactory.cs

```diff
@@ -0,0 +1,58 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Azure.ResourceManager.AppContainers;
+using Azure.ResourceManager.AppContainers.Models;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using ProductConstructionService.Common;
+
+namespace ProductConstructionService.WorkItems;
+public interface IReplicaWorkItemProcessorStateFactory
+{
+    Task<List<WorkItemProcessorState>> GetAllWorkItemProcessorStatesAsync();
+}
+
+public class ReplicaWorkItemProcessorStateFactory : IReplicaWorkItemProcessorStateFactory
+{
+    private readonly ContainerAppResource _containerApp;
+    private readonly IRedisCacheFactory _redisCacheFactory;
+    private readonly IServiceProvider _serviceProvider;
+
+    private List<WorkItemProcessorState>? _states = null;
+
+    public ReplicaWorkItemProcessorStateFactory(ContainerAppResource containerApp, IRedisCacheFactory redisCacheFactory, IServiceProvider serviceProvider)
+    {
+        _containerApp = containerApp;
+        _redisCacheFactory = redisCacheFactory;
+        _serviceProvider = serviceProvider;
+    }
+
+    public async Task<List<WorkItemProcessorState>> GetAllWorkItemProcessorStatesAsync()
+    {
+        if (_states != null)
```

Comment:
What is the point of caching this? It's hard to understand why we have a transient service that caches data and what should be the right use of this class

Reply:
The point was because for the `start` and `stop` endpoint, we'd call it twice, once to perform the action, and then again when we called the `status` endpoint to print out the results. It's changed and removed now